recipes/libpnetcdf/meta.yaml

```diff
@@ -0,0 +1,73 @@
+{% set version = "1.12.3" %}
+{% set build = 0 %}
+
+# recipe-lint fails if mpi is undefined
+{% set mpi = mpi or 'mpich' %}
+{% if mpi == "mpich" %}
+# prioritize mpich via build number
+{% set build = build + 100 %}
+{% endif %}
+
+package:
+  name: libpnetcdf
+  version: {{ version }}
+
+source:
+  url: https://parallel-netcdf.github.io/Release/pnetcdf-{{ version }}.tar.gz
+  sha256: 439e359d09bb93d0e58a6e3f928f39c2eae965b6c97f64e67cd42220d6034f77
+
+build:
+  number: {{ build }}
+  skip: True  # [win]
+  {% set mpi_prefix = "mpi_" + mpi %}
+  # add build string so packages can depend on mpi variants dependencies:
+  # `PKG_NAME * mpi_mpich_*` for mpich
+  # `PKG_NAME * mpi_*` for any mpi
+  string: {{ mpi_prefix }}_h{{ PKG_HASH }}_{{ build }}
+
+  run_exports:
+    - {{ pin_subpackage('libpnetcdf', max_pin='x.x.x') }} {{ mpi_prefix }}_*
+  ignore_run_exports_from:
+    - netcdf-fortran
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('fortran') }}
+    - m4
+    - make
+  host:
+    - {{ mpi }}
+    # these need to be listed twice so conda build picks up the pins
+    - hdf5
```

Comment:
You have overdepending/static linking warnings from hdf5. You either don't need hdf5 or you are static linking (which is not allowed).

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=646866&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=5165

Reply:
It looks like you're right and hdf5 isn't a dependency.  I don't know where I got the misimpression that it was.